ruby-challenges.rb

```diff
@@ -31,3 +105,50 @@
+    #use some sort of iteration over the string?
+        #decision if the letter at the start vs last 2nd vs 2nd to last are ==
+    #or
+    #break the string into an array with each letter at each index
+        #compare current index to index.reverse?
+            #breaking the string into an array means we could check for punctuation and spaces, remove them and still check if the string is a palindrome
+#research of methods in Ruby: .split - converts string to array; each_char-iterates over each character in a string;
+#.chars-converts each character in a string to an array;  
+
+def taco_cat(string)
+    #at first I was getting enumerators as the results because each_char needs to pass |c|-which represents characters through each char to return them in a block? -a block is a loop?
+    if string.downcase.each_char {|c|} == string.downcase.reverse.each_char {|c|} 
+        "#{string} is a tacocat...i mean palindrome!"
+    else
+        "Sorry, #{string} is not a palindrome..."
+    end
+end
+#the problem with the method above would be punctuation and spaces. maybe I can combine it with the .gsub method with a regex. gsub substitutes charcters with something else. maybe I can substitute a deletion? which would be great because then the return would be the original string
+
+p taco_cat(palindrome_tester1) # output --> "Racecar is a tacocat...i mean palindrome!"
+p taco_cat(palindrome_tester2) # output --> "Sorry, LEARN is not a palindrome..."
+p taco_cat(palindrome_tester3) # output --> "Rotator is a tacocat...i mean palindrome!"
+
+def taco_cat2(string)
+                #super cool that [] in regex means exclude, so everything that is not a-z0-9 is being removed! there is a website regex checker that really helped
+    if string.downcase.gsub(/[^a-z0-9]/, '').each_char {|c|} == string.downcase.reverse.gsub(/[^a-z0-9]/, '').each_char {|c|} 
+        "#{string} is a tacocat...i mean palindrome!"
+    else
+        "Sorry, #{string} is not a palindrome..."
+    end
+end
+
+p taco_cat2(palindrome_tester1) # output --> "Racecar is a tacocat...i mean palindrome!"
+p taco_cat2(palindrome_tester2) # output --> "Sorry, LEARN is not a palindrome..."
+p taco_cat2(palindrome_tester3) # output --> "Rotator is a tacocat...i mean palindrome!"
+p taco_cat2("taco Cat") #"taco Cat is a tacocat...i mean palindrome!"
+p taco_cat2("race car race car") #"race car race car is a tacocat...i mean palindrome!"
+p taco_cat2('r4c3c4r')# "r4c3c4r is a tacocat...i mean palindrome!"
+p taco_cat2("red rum, sir, is murder") #"red rum, sir, is murder is a tacocat...i mean palindrome!"
+p taco_cat2("Was it a cat I saw?")#"Was it a cat I saw? is a tacocat...i mean palindrome!"
+p taco_cat2("Sit on a potato pan, Otis.")#"Sit on a potato pan, Otis. is a tacocat...i mean palindrome!"
+p taco_cat2("sator arepo tenet opera rotas")#"sator arepo tenet opera rotas is a tacocat...i mean palindrome!"
```

Comment:
Lol
This is more depth than I have ever seen on palindrome. Well done.

Reply:
Yea i went on a deep dive there haha